File: 1106/addtwo.cpp
// #include <stdio.h>
// #include <iostream>
// //  * Definition for singly-linked list.
// struct ListNode {
//     int val;
//     ListNode *next;
//     ListNode() : val(0), next(nullptr) {}
//     ListNode(int x) : val(x), next(nullptr) {}
//     ListNode(int x, ListNode *next) : val(x), next(next) {}
//  };

class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
    int sum, plus, remain;
    sum = 0;
    plus = 0;
    remain = 0;

    ListNode *returnNode = new ListNode(0);
    ListNode *curr = new ListNode(0);
    int firstroop = 1;
    while (l1!=NULL || l2!=NULL || plus!=0){
        if (l1!=NULL){
            sum += l1->val;
            l1 = l1->next;
        }
        if (l2!=NULL){
            sum += l2->val;
            l2 = l2->next;
        }
        sum += plus;
        plus = sum / 10;
        remain = sum % 10;

        curr->next = new ListNode(remain);
        if (firstroop){
            returnNode = curr;
            firstroop = 0;
        }
        curr = curr->next;
        sum = 0;

    }
    return returnNode->next;

    }

};

// int main(){
//     std::cout << 'aaa' << std::endl;
// }

File: 1106/addtwo.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        sum=0
        plus = 0
        returnNode = cur = ListNode(0)
        while l1 or l2  or plus :
            if l1:
                sum += l1.val
                l1 = l1.next

            if l2:
                sum += l2.val
                l2 = l2.next


            sum = sum + plus
            plus = sum // 10
            remain = sum % 10

            cur.next = ListNode(remain)
            cur = cur.next 
            sum=0

        return returnNode.next


